web-client/src/presenter/actions/validatePetitionDetailsAction.js

```diff
@@ -83,12 +85,30 @@ export const validatePetitionDetailsAction = ({
   if (!errors) {
     return path.success();
   } else {
+    const errorDisplayMap = {
+      statistics: 'Statistics',
+    };
+
+    const {
+      errors: formattedErrors,
+      statisticsErrorMessages,
+    } = aggregateStatisticsErrors({
+      errors,
+      get,
+    });
+
+    const errorMessags = [
+      ...Object.values(omit(errors, 'statistics')),
+      ...statisticsErrorMessages,
+    ];
```

Comment:
while I ❤️ the addition of these error messages, I'm just raising the awareness that this function (and I'm guessing others like it) is getting to be pretty big! do you think there could be some refactoring just to make this code more manageable / establish a leaner pattern?

Reply:
The statistics are kind of a rare beast with regard to how we're doing this. We're aggregating errors for the UI, where there is a dynamic number of statistics that can be added - so some formatting is coming into play here. Furthermore, the way they are returned for the UI isn't 100% in alignment with how they're read in the AlertError component - hence the reason they're being treated differently here. I agree that this isn't ideal, but it's also a bit of an edge case. I think some more attention to cleaning this up is definitely warranted. What do you think about a devex task?

Reply:
+1 [devex task](https://trello.com/c/7XyD3494/960-simplifying-validatepetitiondetailsactionjs)